src/js/actions/tools.js

```diff
@@ -204,15 +207,23 @@ define(function (require, exports) {
             removePromise = Promise.resolve();
         }
 
-        return removePromise.bind(this).then(function () {
-            return this.transfer(policy.addPointerPolicies, pointerPolicyList);
-        }).then(function (policyID) {
-            _currentTransformPolicyID = policyID;
-        });
+        return guidePromise
```

Comment:
would it also work to just `return Promise.join(guidePromise, removePromise).bind(this).then(...` ? 
My gut says that would be easier to read.